### PR TITLE
Commands pr, prs

### PR DESCRIPTION
**Pull request by dbaltas**
* Create a github pull request
* List github open pull requests

### CODE REVIEW

File: cmd/pr.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	prCmd.Flags().StringVar(&description, "description", "", "The description of the PR.")
+	prCmd.MarkFlagRequired("title")
+}
+
+var prCmd = &cobra.Command{
+	Use:   "pr",
+	Short: "Create a pull request [github]",
+	Long:  `Create a pull request on github from compare branch to base branch`,
+	Run: func(cmd *cobra.Command, args []string) {
+		createPR()
```
Comment (mantzas) [inline cmd/pr.go]: you should do some input validation i general. the Command has pre run functions with error PreRunE which can be used for this.

File: cmd/root.go [hunk 2]
```diff
@@ -25,47 +26,57 @@ var (
...
+		noRepoCmds["version"] = true
+
+		if _, ok := noRepoCmds[cmd.Name()]; ok {
+			return
+		}
+
+		err = initializeRepo()
 		if err != nil {
-			fmt.Printf("get repo: %v\n", err)
+			fmt.Printf("Error Initializing repo: %v\n", err)
```
Comment (mantzas) [inline cmd/root.go]: if you use the standard log then log.Fatalf could be used.

File: repo/format.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	outputStrings := []string{}
+	maxLines := 6
+
+	lines := strings.Split(c.Message, "\n")
+	for _, line := range lines {
+		if strings.TrimSpace(line) == "" {
+			continue
+		}
+
+		prefix := ""
```
Comment (mantzas) [inline repo/format.go]: var prefix string is actually preferred.

**Review by mantzas — APPROVED** [review]

File: cmd/pr.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+	},
+}
+
+func createPR() {
```
Comment (nstratos) [inline cmd/pr.go]: It might be better if this function returned an error.

File: cmd/prs.go [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+func init() {
+	rootCmd.AddCommand(prsCmd)
+}
+
+var prsCmd = &cobra.Command{
+	Use:   "prs",
+	Short: "List open pull requests [github]",
+	Long:  `List open pull requests on github`,
+	Run: func(cmd *cobra.Command, args []string) {
```
Comment (nstratos) [inline cmd/prs.go]: It might be preferred to use `RunE` instead of `Run` for better command error handling.

File: cmd/prs.go [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+		branch := *pr.Head.Label
+		if strings.HasPrefix(branch, organizationName+":") {
+			branch = strings.TrimPrefix(branch, organizationName+":")
+		}
+		title := (*pr.Title)
+		if len(title) > 60 {
+			title = title[:60]
+		}
+		t := *pr.CreatedAt
+		at := fmt.Sprintf("%d-%02d-%02d %02d:%02d", t.Year(), t.Month(), t.Day(), t.Hour(), t.Minute())
```
Comment (nstratos) [inline cmd/prs.go]: Check if [t.Format](https://golang.org/pkg/time/#Time.Format) makes this line easier.

File: cmd/root.go (hunk 2 above)
Comment (nstratos) [reply cmd/root.go]: Or `fmt.Fprintf(os.Stderr, ...)` since `os.Exit` is called explicitly.

File: cmd/root.go [hunk 7]
```diff
@@ -87,29 +98,29 @@ func initConfig() {
...
+		r, err = repo.NewFromPath(path, viper.GetString("generic.remote"))
 	}
 
-	rmt, err := repository.Remote(viper.GetString("generic.remote"))
-
 	if err != nil {
-		fmt.Println(err)
-		return nil, err
+		fmt.Printf("Error loading repo:%s\n", err)
+		return err
```
Comment (nstratos) [inline cmd/root.go]: Consider `return fmt.Errorf("error loading repo: %s\n", err)`.

File: cmd/version.go [hunk 8]
```diff
@@ -15,6 +15,6 @@ var versionCmd = &cobra.Command{
 	Short: "Print the version of ergo",
 	Long:  `Print the version of ergo`,
 	Run: func(cmd *cobra.Command, args []string) {
-		fmt.Println("0.2.1")
+		fmt.Println("0.2.2")
```
Comment (nstratos) [inline cmd/version.go:18]: Consider embedding the version in the binary when issuing `go build` like this:

    go build --ldflags="-X main.version=0.2.2"

In this case, `version` has to be a global variable in main. The version can be taken by the git tags with something like `git describe --tags`. Here's how I've done it in [make.go](https://github.com/nstratos/make.go/blob/master/make.go#L95).

When we get vgo in go1.12 this procedure will hopefully be handled by the go tooling and there will be no need for all this.

File: repo/repo_functional_test.go [hunk 9]
```diff
@@ -5,49 +5,45 @@ import (
...
 
-	r := New(repoURL, directory, "origin")
+	path := "/tmp/ergo-functional-test-repo"
+	repoURL := "https://github.com/dbaltas/ergo-functional-test-repo.git"
+
+	// r := New(repoURL, path, "origin")
 	// cleanup after test run
 	defer func() {
-		os.RemoveAll(directory)
+		os.RemoveAll(path)
```
Comment (nstratos) [inline repo/repo_functional_test.go]: Missing `err = os.RemoveAll(path)`.

File: repo/repo_functional_test.go [hunk 10]
```diff
@@ -5,49 +5,45 @@ import (
...
-		if err != nil || repoFromClone == nil {
-			t.Errorf("Error cloning repo:%v\n", err)
-			return
-		}
-	})
+	r, err := NewClone(repoURL, path, "origin")
+
+	if err != nil || r == nil {
+		t.Errorf("Error cloning repo:%v\n", err)
+		return
```
Comment (nstratos) [inline repo/repo_functional_test.go]: No need for `t.Errorf` and `return`. Use `t.Fatalf` instead.

**Review by nstratos — APPROVED** [review]

File: cmd/prs.go (hunk 6 above)
Comment (dbaltas) [reply cmd/prs.go]: I tried t.Format("2018-04-23 17:54") but didn't get the expected result

File: cmd/version.go (hunk 8 above)
Comment (dbaltas) [reply cmd/version.go:18]: sounds interesting, will look into it

File: repo/repo_functional_test.go (hunk 9 above)
Comment (dbaltas) [reply repo/repo_functional_test.go]: 👍

File: repo/format.go (hunk 3 above)
Comment (dbaltas) [reply repo/format.go]: 👍

File: cmd/prs.go (hunk 6 above)
Comment (nstratos) [reply cmd/prs.go]: Try `t.Format("2006-01-02 15:04")`. You can find the layout from the [time constants](https://golang.org/pkg/time/#pkg-constants).

Comment (dbaltas) [reply cmd/prs.go]: Thanks!

**Pull request merged**